doc/barclamp.md

```diff
@@ -1356,3 +1356,147 @@ place) you will have to add additional setup code to
+
+# pacemaker definitions
+   [:api][:op][:monitor][:interval] = "10s"
+   [:worker][:op][:monitor][:interval] = "10s"
+   [:keystone_listener][:op][:monitor][:interval] = "10s"
+
+   [:worker][:agent] = "systemd:openstack-barbican-worker"
+   [:keystone_listener][:agent] = "systemd:openstack-barbican-keystone-listener"
+
+```
+> - Resource Agent: In 'LayMan' terms, resource agent is a script that the
+    cluster manager uses to manage the resource.
+> - LSB: is a specification of writing resource agent. 
+> - OCF: Pacemaker specific extension to LSB.
+> - systemd: Use systemd unit files for managing the resource.
+
+- [barbican_service.rb(cookbook)](https://github.com/crowbar/crowbar-openstack/commits/master/chef/cookbooks/barbican/definitions)
+
+   Makes the installation of the service HA aware, by using [```provider  Chef::Provider::CrowbarPacemakerService```](https://github.com/crowbar/crowbar-ha/blob/master/chef/cookbooks/crowbar-pacemaker/providers/service.rb)
+(Beware internal documentation on link) By adding that line we tell chef to
+use **service resource** from that provider and not the default one. The file has huge documentation
+and is good read for some internal interactions between chef and pacemaker. Chef tries to manage
+resources locally, while pacemaker operations are cluster-wide. So we need special ways
+to interact with pacemaker about status of clone/group/service and its configuration.
+
+
+- [helpers.rb](https://github.com/crowbar/crowbar-openstack/commits/master/chef/cookbooks/barbican/libraries/)
+
+    Here we define a helper class to initialize the variables/ports correctly ```if ha_enabled```
+
+- [api.rb](https://github.com/crowbar/crowbar-openstack/commits/master/chef/cookbooks/barbican/recipes/api.rb)
+
+   First make sure the service and user are correctly registered witht keystone, use keystone_setting_helper to get keystone endpoint details. Then register the endpoint with keystone, using appropriate port.
+	```
+	If !ha => use the standard port
+	else   => use port where ha is listening
+	...
+	bind api on correct port accordingly
+	```
+    If using ha, host address will be from admin network, as HAProxy will listen
+    virtual-ip-address form admin network, as the node is part of the cluster.
+    Otherwise just register on '*'
+
+- [common.rb](https://github.com/crowbar/crowbar-openstack/commits/fb4cd02a09ba35182e401cac48858e2e5d1d5508/chef/cookbooks/barbican/recipes/common.rb)
+
+   Common recipe is executed for all services and on all nodes of a cluster in
+   parallel. So in case of ha, the db/user should only be created once, so make
+   sure only ```cluster_founder``` allowed to do these tasks and create sychronization
+   markers around them.  Also be sure to listen for request to VIP, as request will be
+   sent to VIP in case of HA.(which is actually the cluster address where HAproxy is
+   listening)
+    
+
+
+
+> To make sure the other nodes do not start accessing the database before the
```

Comment:
I think you probably don't want the leading `>` here, as that has a specific meaning in GH-flavoured markdown.